### PR TITLE
Use axios instead of rpn

### PR DESCRIPTION
**Pull request by mykolab-wix**
1. `request-promise-native` is a year as deprecated
2. it leads to errors in Jest if `request-promise` is used by Serverless user
3. it uses `stealthy-require` which is pretty hard stuff and may lead to nasty bugs in runtime for Serverless

### CODE REVIEW

**Comment by bildja** [comment]
@lelikg I would suggest that Axios wouldn't get into the client SDK, it's already big enough.

**Comment by lelikg** [comment]
> @lelikg I would suggest that Axios wouldn't get into the client SDK, it's already big enough.

@bildja it's not my PR, but I think having axios has more benefit than downsides.
@mykolab-wix who requested this PR?

File: src/platform/http/http-client.ts [hunk 1]
```diff
@@ -60,46 +56,41 @@ export interface IHTTPClient {
...
     }
 
-    return request(options).then((response) => {
-      if (response.statusCode < 200 || response.statusCode >= 300) {
-        return Promise.reject(new Error(JSON.stringify(response.body)));
-      }
-
-      return response.body;
-    });
+    return await axios(options)
```
Comment (lelikg) [inline src/platform/http/http-client.ts:81]: why `await` here?

File: src/platform/http/http-client.ts [hunk 2]
```diff
@@ -113,25 +104,17 @@ export class HTTPClient implements IHTTPClient {
...
-          return Promise.reject(response.body);
-        }
-
-        return response;
-      },
-      (error) => {
-        return Promise.reject(error);
-      },
-    );
+    return await axios(options)
```
Comment (lelikg) [inline src/platform/http/http-client.ts:113]: why await?

**Review by lelikg — APPROVED** [review]

**Comment by bildja** [comment]
@lelikg I mentioned you because you introduced axios recently and we talked about it, now it's spreading.
As for "who requested" – Mykola contacted me if we are planning to replace request-promise-native with something because it causes conflicts for them and I suggested making a PR

File: src/platform/http/http-client.ts (hunk 1 above)
Comment (mykolab-wix) [reply src/platform/http/http-client.ts:81]: To have nice stack trace in case of error

File: src/platform/http/http-client.ts (hunk 2 above)
Comment (mykolab-wix) [reply src/platform/http/http-client.ts:113]: To have nice stack trace in case of error

File: src/platform/http/http-client.ts (hunk 1 above)
Comment (mykolab-wix) [reply src/platform/http/http-client.ts:81]: I have no hard preference though, it's your code, we just need to get rid of request-promise-native

**Comment by lelikg** [comment]
I think that we can live with Axios and to be honest, the client sdk should be separate from server sdk

**Comment by bildja** [comment]
100% agree that client SDK should be separate from the server, but now it's not the case. 
And I was asking for it not to be in the client as a consumer, not as a maintainer :)

**Comment by mykolab-wix** [comment]
@lelikg Serverless team requested it, as some of our users have conflicts during testing, [see](https://github.com/request/request-promise/issues/247)

**Pull request merged**